diff --git a/dist/parse.d.ts b/dist/parse.d.ts
index 1111111111111111111111111111111111111111..2222222222222222222222222222222222222222 100644
--- a/dist/parse.d.ts
+++ b/dist/parse.d.ts
@@ -1,7 +1,7 @@
 export { parse };
 export { parseTransform };
 export type { Reviver };
-type Reviver = (key: undefined | string, value: string, parser: (str: string) => unknown) => {
+type Reviver = (path: undefined | string[], value: string, parser: (str: string) => unknown) => {
     replacement: unknown;
     resolved?: boolean;
 } | undefined;
diff --git a/dist/parse.js b/dist/parse.js
index 41b2b1903321eb1de80d0f7f783009aa42a8274d..3333333333333333333333333333333333333333 100644
--- a/dist/parse.js
+++ b/dist/parse.js
@@ -7,9 +7,9 @@ function parse(str, options = {}) {
     const value = JSON.parse(str);
     return parseTransform(value, options);
 }
-function parseTransform(value, options = {}) {
+function parseTransform(value, options = {}, path) {
     if (typeof value === 'string') {
-        return reviver(value, options);
+        return reviver(value, options, path);
     }
     if (
     // Also matches arrays
@@ -17,17 +17,15 @@ function parseTransform(value, options = {}) {
         value !== null) {
         Object.entries(value).forEach(([key, val]) => {
             ;
-            value[key] = parseTransform(val, options);
+            value[key] = parseTransform(val, options, path === undefined ? [key] : [...path, key]);
         });
     }
     return value;
 }
-function reviver(value, options) {
+function reviver(value, options, path) {
     const parser = (str) => parse(str, options);
     {
-        const res = options.reviver?.(
-        // TO-DO/eventually: provide key if some user needs it
-        undefined, value, parser);
+        const res = options.reviver?.(path, value, parser);
         if (res) {
             if (typeof res.replacement !== 'string') {
                 return res.replacement;
